### DISCUSSION TITLE
is there a powershell equivalent?

### DISCUSSION THREAD

**Discussion by minoseah629** [PowerShell/PowerShell#16931] [Q&A] [answered]
is there an equivalent capability in powershell to do this in c#: 

`client.GetResource<Dictionary<string, object>>("1234-wxyz");`

i tried `client.GetResource[Dictionary[string, object]]("1234-wxyz")`
but got array index expression is missing or not valid

**Comment by iSazonov** [comment]
As quick workaround you could use C# code and Add-Type.

**Comment by vexx32** [comment, marked as answer]
If you're on the latest preview I think the PR for generic method calls  may already be in that version, test it out and see if it does the job. 🤔 

If not, or you need it to work in an earlier version, you'll need to reach through the Reflection APIs to create a generic method to call directly. From memory it's something along the lines of:

```ps1
$type = $client.GetType()
$method = $type.GetMember('GetResource')[0]
$method = $method.MakeGenericMethod([Dictionary[string, int]])
$method.Invoke("1234-wxyz")
```

And as Ilya mentioned, invoking C# directly is an option as well.

**Reply by minoseah629** [reply]
thank you @vexx32.  this worked out in 7.3.0-preview.2 on mac using what tried in the initial post

`client.GetResource[Dictionary[string, object]]("1234-wxyz")`

by chance, do you have that pr link? would like to check it out

**Reply by vexx32** [reply]
Aye, it's this one.

https://github.com/PowerShell/PowerShell/pull/12412

I think there's a follow up issue for an edge case in the parser I need to look at soon.
